### PR TITLE
`babylon`: fix error location for class properties with a missing sem…

### PR DESCRIPTION
**Pull request by hzoo**
…icolon

Ref https://phabricator.babeljs.io/T6925

### CODE REVIEW

File: packages/babylon/src/parser/statement.js [hunk 1]
```diff
@@ -748,7 +748,7 @@ pp.parseClassProperty = function (node) {
     node.value = null;
   }
   if (!this.eat(tt.semi)) {
-    this.raise(this.state.start, "A semicolon is required after a class property");
+    this.raise(node.value && node.value.end || node.key.end, "A semicolon is required after a class property");
```
Comment (amasad) [inline packages/babylon/src/parser/statement.js]: is node.key guaranteed to be set? Should we fallback to `this.state.start`?

Comment (hzoo) [inline packages/babylon/src/parser/statement.js]: It should be? for `asdf = 1;` value is set and for `asdf;` key is set

**Comment by ngduc** [comment]
+1 for standard-js (no semicolons)

**Comment by ruifortes** [comment]
also +1 for standard-js (no semicolons)

**Comment by hzoo** [comment]
6.5.2 will revert these PRs

**Pull request merged**